batchflow/batch.py

```diff
@@ -493,22 +494,69 @@ def apply_transform(self, ix, func, *args, src=None, dst=None, p=None, use_self=
 
             for item in range(len(batch)):
                 self.dst[item] = func(self.src[item], *args, **kwargs)
+
+        Examples
+        --------
+
+        ::
+
+            apply_transform(make_masks_fn, src='images', dst='masks')
+            apply_transform(apply_mask, src=('images', 'masks'), dst='images', use_self=True)
+            apply_transform_all(rotate, src=['images', 'masks'], dst=['images', 'masks'], p=.2)
         """
+        to_act = bool(p is None or np.random.binomial(1, p))
+
+        if isinstance(src, list) and len(src) == len(dst):
+            return tuple([self._apply_transform(ix, func, to_act, *args, src=item, use_self=use_self, **kwargs)
+                          for item in src])
```

Comment:
`component` would look much better than `item`

Reply:
doesn't enter this condition if `dst` is not provided. `dst` must be equal to `src` if not provided.

Reply:
also doesn't work if src is a list of length 1.

Reply:
And this condition is passed if `src` is a list, `dst` is str and `len(src)` happens to be equal to `len(dst)`.

Reply:
line 560 in _apply_transform src cannot be list here because you checked it in apply_transform, if i am not mistaken